controllers/opentelemetrycollector_controller.go

```diff
@@ -75,6 +76,22 @@ type Params struct {
 	Config   config.Config
 }
 
+func (r *OpenTelemetryCollectorReconciler) Validate(otelcol v1beta1.OpenTelemetryCollector) admission.Warnings {
```

Comment:
Is there any reason for this to be a method on the reconciler? Looks like the only thing we're using from it is the config, and the webhook has access to that anyway. I'd just make it a function in `collector_webhook.go`. That would also let us avoid the setup changes.

Reply:
we can't do it in the webhook because then we would have an import cycle (through the collector.Build) unfortunately, previously this was an anonymous function in the main.go file, but it felt a bit odd to put it there IMO.

Reply:
Ah, makes sense. Honestly, I think I'd like it better as an anonymous function in main, or in some other package. I don't like adding it to the reconciler, which is completely unrelated functionally, and I don't really like creating a runtime dependency between it and the webhook.

That said, I do want to merge this PR, so if you and @taniyourstruly feel strongly about keeping it the way it is, I can accept it.

Reply:
i think we can make it an anon func or just a named func in main.